### PR TITLE
Night mode version for docs

### PR DESCRIPTION
**Pull request by pdilyard**
This could probably use some cleaning up, but this is a good starting point. Here are some screenshots:

Feel free to pull it down and play with it, I'd love to hear any feedback.

Solves #474

### CODE REVIEW

**Comment by josevalim** [comment]
Thank you @pdilyard, this looks great! <3 Some points to discuss:

1. Can we make the font less white and a bit more greyish? I am afraid the white is a bit too strong right now and won't be beneficial on night mode. Tim's script had this: https://pbs.twimg.com/media/CTqMbN5UAAAqO0b.png:large

2. I am not convinced about the "day | night" switch at the top. It is a command that won't be triggered frequently. I was wondering we could have it at the footer of the page. Something like:

```
Built using ExDoc (v0.10.0),  designed by Friedel Ziegelmayer.
              Choose: Day mode | Night mode
```

However, please don't act on this feedback yet. Let me know your opinion and let's also wait for @Dignifiedquire's input.

Everything else looks excellent to me. Great use of local storage!

**Comment by pdilyard** [comment]
Thanks for the quick response @josevalim.

1. You're probably right about the colors not being beneficial in night mode. I went with as many predefined colors as possible, but I could definitely add a special color palette for the night fonts.

2. I really like the idea of the switch in the footer a lot better than the button I have now.

File: assets/less/night/content/functions.less [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+.docstring {
+
+  h2,
+  h3,
+  h4,
+  h5 {
+
+  }
+
+  h2 {
```
Comment (milmazz) [inline assets/less/night/content/functions.less]: This block can be deleted.

File: assets/less/night/content/functions.less [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  }
+
+  h3 {
```
Comment (milmazz) [inline assets/less/night/content/functions.less]: This block can be deleted.

File: assets/less/night/content/functions.less [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  }
+
+  h4 {
```
Comment (milmazz) [inline assets/less/night/content/functions.less]: This block can be deleted.

File: assets/less/night/content/functions.less [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+  }
+
+  h5 {
```
Comment (milmazz) [inline assets/less/night/content/functions.less]: This block can be deleted.

File: assets/less/night/content/general.less [hunk 5]
```diff
@@ -0,0 +1,10 @@
+h1, h2, h3, h4, h5, h6 {
```
Comment (milmazz) [inline assets/less/night/content/general.less]: Just to be consistent with the rest of the code, please, use one line per header:

```css
h1,
h2,
h3,
h4,
h5,
h6 {
...
}
```

File: assets/less/night/content/functions.less [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+.specs {
+  .spec-title {
+
+  }
+
+  .specs-list {
+
+    pre {
+
```
Comment (milmazz) [inline assets/less/night/content/functions.less]: :scissors: Extra space

File: assets/less/variables.less [hunk 7]
```diff
@@ -17,6 +17,12 @@
 
 @detailGray: #f7f7f7;
 
+// Night mode
+@nightGray: #1A1D26;
+@nightCodeBlue: #4B556E;
+@nightMediumBlue: #69778C;
+@nightPurple: #CEB4ED;
+
```
Comment (milmazz) [inline assets/less/variables.less]: :scissors:

File: assets/less/night/content/functions.less (hunk 1 above)
Comment (pdilyard) [inline assets/less/night/content/functions.less]: Oops, forgot to take all these extras out :smile:

File: assets/less/variables.less (hunk 7 above)
Comment (pdilyard) [inline assets/less/variables.less]: Is this comment a mistake? I'm using those variables. Or are you suggesting I move them to a variables.less file inside the night folder?

Comment (josevalim) [inline assets/less/variables.less]: I think it was regarding the extra new line or whitespace :D

Comment (milmazz) [inline assets/less/variables.less]: Sorry, I just meant to "delete the extra space" :)

**Comment by pdilyard** [comment]
@Dignifiedquire ping

**Comment by Dignifiedquire** [comment]
will take a look in the morning

**Comment by Dignifiedquire** [comment]
@pdilyard thanks for this, some feedback below:

* Switch, it looks great, but the position is not great, so if we can just move it as it is into the footer?
* Colors, we definitely need to change some things here. Most important are
  * Code border and background
  * Code syntax theme should be one that's designed for dark mode (I suggest tomorrow night from highlight.js)
  * Text color in general (probably should be lighter)
  * Function header background

**Comment by pdilyard** [comment]
Alright here are some updates of where I'm at with @Dignifiedquire's feedback so far.

- I changed the syntax theme to `tomorrow-night` in night mode.
- I changed the background of the code blocks to a dark gray that works a little better with the syntax highlighting.
- I changed the overall text color to a less-harsh gray.
- I ended up just moving the switch to the footer as a text link. I tried to move the icon-based switch as it was to the footer, but I couldn't figure out a good way to position it that I felt was easy to find.

Note the new text colors:


Note the new toggle switch in the footer, as well as the new code blocks:


I also have a question for @Dignifiedquire:
- What were you thinking for the function header backgrounds? Should those be more gray? I want to make sure they stand out, as discussed in #487.

Any other feedback is much appreciated :smile:

**Comment by josevalim** [comment]
This looks great. Once again, let's wait for @Dignifiedquire to sign off and merge this. :) We should do a release right after! This will be great with the other style changes.

**Comment by Dignifiedquire** [comment]
Sorry guys for being so absent, checking the branch out now

**Comment by Dignifiedquire** [comment]
@pdilyard are you sure you pushed your changes, pulling down your branch there seem to be no changes

**Comment by pdilyard** [comment]
@Dignifiedquire I don't know what just happened here. For some reason git thought I needed to pull down my feature branch and gave me a bunch of merge conflicts on it, even though the feature branch has only ever had changes made by me directly. I resolved the conflicts, and it threw in a bunch of repeat commits. Let me know if you have a tip on how I can clean up this history a bit.

**Comment by josevalim** [comment]
@Dignifiedquire nothing to be sorry about :D I mostly mention you to be make it clear to @pdilyard why we are waiting. It is OK to be absent or to tell us to wait for a week or even to tell us to do as we please if you can't take a look at it soon. :)

**Comment by josevalim** [comment]
@pdilyard it is ok if the branch is a bit messy, I can squash everything before merging. Just make sure there is no spurious code in the diff here on Github and we are fine. :)

**Comment by pdilyard** [comment]
@josevalim thanks, the diff looks good :+1:

**Comment by pdilyard** [comment]
But anyway, @Dignifiedquire the changes should be there for you now

**Comment by Dignifiedquire** [comment]
Alright, I think this is in a really good state now, only thing I'm not a 100% happy with is the function header background and the text background, but I don't have any good ideas right now I'm afraid

**Comment by josevalim** [comment]
Ok. I am going to merge and do a new release today. :) We will wait for more feedback and continue improving when it comes to the background colors.

**Comment by josevalim** [comment]
Thank you @pdilyard and @Dignifiedquire!  :heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**